src/odoc/rendering.ml

```diff
@@ -128,22 +115,29 @@ let generate_source_odoc ~syntax ~warnings_options ~renderer ~output
   | Page_content _ | Unit_content _ | Asset_content _ ->
       Error (`Msg "Expected an implementation unit")
 
+let generate_asset_odoc ~warnings_options:_ ~renderer ~output ~asset_file
+    ~extra_suffix extra file =
+  Odoc_file.load file >>= fun unit ->
+  match unit.content with
+  | Odoc_file.Asset_content unit ->
+      let doc = Renderer.document_of_asset asset_file unit in
+      render_document renderer ~output ~sidebar:None ~extra_suffix ~extra doc;
```

Comment:
There was a lot of simplification in this PR, why not continue and remove assets from the document and the renderer ?
`document_of_asset` seems very orthogonal to the rest of the document (it's just the `Asset` constructor which shares nothing with the other document types) and because of `render_document` and the `renderer` type, the content of the asset file has to go through a `Format` buffer, which seems totally unnecessary.
I think all of this could be removed.

Reply:
True! (I started that and decided to stop, thinking you would not see it, so I stashed everything. Time to stash pop!)